configure.ac

```diff
@@ -614,15 +614,7 @@ AC_ARG_ENABLE(
 #####################################################
 AM_CONDITIONAL([BUILD_PYTHON_LIBS], [test "x$enable_python_libs" = xyes])
 
-AS_IF([test "${enable_python_libs}" = "yes"],
-      [AM_PATH_PYTHON(2.6)
-       AS_IF([test -z $PYTHON], [PYTHON="python"])
-       PYTHON_NAME=`basename $PYTHON`
-       AC_CACHE_CHECK([for $PYTHON_NAME module: google.protobuf],
```

Comment:
I think we should really try and keep the protobuf Python check somehow.

Reply:
The main difference between the Debian package and a build that uses a setup.py is that the Debian package installs the lib globally, and one that uses a setup.py can be installed in a virtualenv or globally.

Reply:
If the python libraries continue to be shipped with the rest of ola, I think it would be preferable to also move the python libraries to a separate source archive (possibly also a separate git repo, but I'll leave that for you to decide).

I think it would be confusing to say "if you want to use the python libraries, you need to run `cd foo; python setup.py` in addition to whatever else you need to do to compile things". After all, if we're going to do this for python, then why stop there? Might want to also say "if you want the C++ libraries, please do `cd libs; ./configure; make`", etc, and that way lies madness.

Reply:
Oh yes. As to what I'd do for Debian: if this goes through I guess I would have to switch to it, but it would make things very very complicated, since I have to now add multiple override targets to tell `dh` that this is a very odd build system.

I'd rather not have to do that.

Reply:
Thinking about this broadly, given we only allow localhost connections to olad, what's the benefit of installing the Python client via one means and olad via another (aside from allowing a Python app to express it's dependency on the Python client within PyPi)?

So you're broadly in favour of staying as is @yoe ? Or you'd favour the Python libs going to a separate archive/git repo regardless? We'd have to do some submodule stuff or something, because the clients all rely on building against the proto definitions (and indeed the Art-Net and USB Pro proto definitions).

I wonder if we need to consider this as parallel options; ./configure is the core/base option, with Debian, MacPorts, RPM, PyPi etc being alternative ways to install some or all of the software, either using ./configure directly, or tracking other things in parallel (we'd have to update setup.py when we add a new Python dependency, in the same way as we update Debian's control file). Likewise just because we ship an RPM spec, we don't expect Debian to Alien that to make it's packages (and indeed the same way the Debian release differs from a .deb built directly from our git repo, due to licensing considerations around jQuery etc).

Do we just need to define that ./configure is the definitive version?

Reply:
I guess it depends how strictly you take single build system @yoe . @simark has made it so Python's setup.py gets called from the Makefile:
https://github.com/OpenLightingProject/ola/pull/772/files#diff-497fcf63005c140b5833f112a7ad1d14R5

Although I guess that would bypass some of the prefix type settings you might pass to configure? The Java library build already calls out to Maven to make their libs.

I think from my understand it wouldn't fail, as it would pull in dependencies from pip automagically, although that means they aren't managed via your primary package manager (apt or similar), although again Java does the same thing already.

@simark I think I still feel the same as I did initially, configure should be the master, so should retain all the checks, but I've no objection to shipping parallel means of packaging all or part of the system within the main codebase (i.e. setup.py, debian/control etc). Yes it's a bit of duplication, but we've got some of that already, but it also makes it nice and obvious on a standalone system what might happen. Personally I try and use the system's primary package management system for everything (Perl, Python etc), so I'd find it irritating if something goes off in the background and installs other copies of libraries.